contracts/src/subnet/SubnetActorCheckpointingFacet.sol

```diff
@@ -153,4 +96,28 @@ contract SubnetActorCheckpointingFacet is SubnetActorModifiers, ReentrancyGuard,
             revert InvalidSignatureErr(uint8(err));
         }
     }
+
+    /// @notice Ensures the checkpoint is valid.
+    /// @dev The checkpoint block height must be equal to the last bottom-up checkpoint height or
+    /// @dev the next one or the number of bottom up messages exceeds the max batch size.
+    function ensureValidCheckpoint(BottomUpCheckpoint calldata checkpoint) internal view {
+        uint64 maxMsgsPerBottomUpBatch = s.maxMsgsPerBottomUpBatch;
+        if (checkpoint.msgs.length > maxMsgsPerBottomUpBatch) {
+            revert MaxMsgsPerBatchExceeded();
+        }
+
+        // if the bottom up messages' length is max, we consider that epoch valid
+        if (checkpoint.msgs.length == s.maxMsgsPerBottomUpBatch) {
+            return;
+        }
+
+        // the max batch size not reached, we only support checkpoint period submission.
+        uint256 lastBottomUpCheckpointHeight = s.lastBottomUpCheckpointHeight;
```

Comment:
I guess this is fine for now, but we may relaxing this requirement in the future and let children submit a checkpoint whenever they want or need to.